### ISSUE TITLE
Output image not generating

### ISSUE THREAD

**Issue by jsmentch** [Telecommunication-Telemedia-Assessment/GBVS360-BMS360-ProSal#3]
Thanks for making this code accessible!

I am working on a windows computer and am having trouble getting images to export. As a very simple test, I've moved a test equirectangular image into the _\GBVS360-BMS360-ProSal\x64\Debug_ directory

Using PowerShell in this same directory I am running the command:
_.\Salient360.exe -i .\test.jpg -o .\testout.jpg --general-model 2_

No errors are generated but no output image in this directory is exported. Am I using this command incorrectly or might there be something else going wrong? Thank you!

**Comment by plebreton** [comment]
Hello,

That is strange, you use it correctly and it should work.

- At first, maybe try to use one of the pre-built version: https://github.com/Telecommunication-Telemedia-Assessment/GBVS360-BMS360-ProSal/releases/tag/2.0
- If it works, I am wondering if it is due to the ffmpeg dll that should be next to the binary. That DLL should be in the opencv.


- If it still does not work, the fastest way to identify the problem would be track the execution using visual studio: 

Open Visual Studio, go to the "solution explorer" tab, look for "Salient360", right click on that project, go to properties. In the Salient360 Property pages, Go to Configuration properties, Debugging, there you have a "command line argument". There, add -i .\test.jpg -o .\testout.jpg --general-model 2. Save & close that window.

Now you can tack the execution in the code. Add a break point at line 600 in main.cpp of the Salient360 project (click on the left of the number lines, that should add a red point). Then, hit F5, and you should be able to monitor the execution. Use F10 to go to next steps, and check where it stops. 

 Maybe also try from the simple shell (cmd), or by compiling the project in release mode, or by using the image in the img/equirectangular folder. But I would just suggest to use VS studio to find the problem, it will be much quicker. 


Edit : ah, and I noticed you said that you cannot "export", does it mean that you can see the result if you do not provide an output file? (without the option - o)

**Comment by jsmentch** [comment]
Great! I didn't realize there was a pre-built version. Yes I was not seeing results even if i did not provide an output file.

This pre-built version is working with the basic options --general-model 1 and 2, however I cannot use other options. I would like to try the GBVS360 with DIO channels for example, so I run:

.\salient.exe -i .\test.jpg -o .\testoutDIO.jpg --general-model 2 --gbvs360-channels DIO

and receive this error:
Error incorect program options. See --help...

I also see the help in this version does not mention an equatorial prior option? Is this excluded by default? I would like to use the GBVS360 with no equator prior added. Thank you!

**Comment by plebreton** [comment]
Ok, I see. 

I think it may be the image codec which are missing. Check your opencv lib folder, and find the ffmpeg and image codec DLL, then put them next to the binary and you should be able to make it run.

The build indeed have a limited number of arguments, as it was meant to be easy to use for both the organizer of the grand challenge and anybody who will use this application in the future. So it is easy to run repeatable comparisons. 

I can also see that the comments in the DISABLED command line argument are not accurate. You would better have a look in the code, and more precisely in that class: https://github.com/Telecommunication-Telemedia-Assessment/GBVS360-BMS360-ProSal/blob/master/libgbvs360/Saliency360.cpp
To really see what parameters was used. For example, GBVS was configured using the channels DIO. Parameters for BMS360 are also wrong in the DISABLED manual. 

If you want to investigate the differences with/without equatorial prior, I made a build there: https://github.com/Telecommunication-Telemedia-Assessment/GBVS360-BMS360-ProSal/releases/tag/0

Please be careful with what you do with it, as model performance depends on the configuration... 

Finally, as reported in the paper: https://www.sciencedirect.com/science/article/abs/pii/S0923596518302406

GBVS360 is not really the best approach, you should better use BM360. 


In the build I provided above, you can find GBVS360 in the legacy-icme-2017 models. When you use the "general model 2". It is quite easily recognizable with all the logs about projection, computation per projection, back projection, activation etc. 

The BMS360 model being the one enabled with you don't use the legacy-icme-2017 option. 

In summary: 

# BMS360
.\salient.exe -i .\test.jpg -o .\test.jpg --general-model 1 
.\salient.exe -i .\test.jpg -o .\test.jpg --general-model 2

# BMS
.\salient.exe -i .\test.jpg -o .\test.jpg --general-model 1 --legacy-icme-2017

# GBVS360
.\salient.exe -i .\test.jpg -o .\test.jpg --general-model 2 --legacy-icme-2017


And in each case, you can also add the equatorial prior with the option --equatorial-prior

I will make a more compressive command line tool at a later point, as I don't have the time right now.

**Comment by jsmentch** [comment]
Okay perfect, that makes sense. Thank you so much for the info and for your support with this!!
